jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -781,14 +782,27 @@ void onTrailers(HttpFields trailers)
 
     public void onCompleted()
     {
-        HttpServletRequest httpServletRequest = _request.getHttpServletRequest();
+        ServletContextRequest.ServletApiRequest apiRequest = _request.getServletApiRequest();
         if (LOG.isDebugEnabled())
-            LOG.debug("onCompleted for {} written={}", httpServletRequest.getRequestURI(), getBytesWritten());
+            LOG.debug("onCompleted for {} written={}", apiRequest.getRequestURI(), getBytesWritten());
 
         long idleTO = _configuration.getIdleTimeout();
         if (idleTO >= 0 && getIdleTimeout() != _oldIdleTimeout)
             setIdleTimeout(_oldIdleTimeout);
 
+        if (getServer().getRequestLog() != null)
+        {
+            Authentication authentication = apiRequest.getAuthentication();
+            if (authentication instanceof Authentication.User userAuthentication)
+                _request.setAttribute(CustomRequestLog.USER_NAME, userAuthentication.getUserIdentity().getUserPrincipal().getName());
+
+            String realPath = apiRequest.getServletContext().getRealPath(_request.getPathInContext());
+            _request.setAttribute(CustomRequestLog.REAL_PATH, realPath);
+
+            String servletName = _request.getServletName();
+            _request.setAttribute(CustomRequestLog.HANDLER_NAME, servletName);
```

Comment:
This all seems like something we used to do in a Customizer.

Reply:
All these attributes can only be set after the request has been handled, just before exiting the Servlet handling.
In a `Customizer` we don't have access to the Servlet name (nor the other values we store in these attributes), so they must really be done here.

Reply:
Fair enough